server/channels/app/plugin_api.go

```diff
@@ -1262,3 +1263,34 @@ func (api *PluginAPI) GetUploadSession(uploadID string) (*model.UploadSession, e
 	}
 	return fi, nil
 }
+
+// SendPushNotification will attempt to send a push notification to `notification.User`, using
+// `notification.Post` as the source of the notification. Note: the NotificationWillBePushed hook will
+// be run after SendPushNotification is called.
+func (api *PluginAPI) SendPushNotification(notification *model.PluginPushNotification) error {
+	var profiles map[string]*model.User
+	var err error
+	if notification.Channel.Type == model.ChannelTypeGroup {
+		if profiles, err = api.app.Srv().Store().User().GetAllProfilesInChannel(context.Background(), notification.Channel.Id, true); err != nil {
+			return err
+		}
+	}
+
+	sender, appErr := api.app.GetUser(notification.Post.UserId)
+	if appErr != nil {
+		return appErr
+	}
+	user, appErr := api.app.GetUser(notification.UserID)
+	if appErr != nil {
+		return appErr
+	}
+
+	postNotification := &PostNotification{
+		Post:       notification.Post,
+		Channel:    notification.Channel,
+		ProfileMap: profiles,
+		Sender:     sender,
+	}
+	api.app.sendPushNotification(postNotification, user, false, false, "")
```

Comment:
I don't know what `false`, `false`, and `""` are, but also wondering if plugins might want to customize these or other parameters? See comments in parent PR, but wondering if we can just accept the full `model.PostNotification` here and backfill anything in that the plugin leaves unset?

Reply:
Those are `explicitMention`, `channelWideMention`, and `replyToThreadType` -- first two are for generating the default messages when the server's `cfg.EmailSettings.PushNotificationContents` is set to certain values (or the server isn't fully licensed), the last is for CRT thread replies. We could definitely accept those as well in the `model.PluginPushNotification`. Let me make that change.

I'm not against accepting the `model.PostNotification`, as it would simplify a lot of the contextual knowledge you need when figuring out how to make a push notification from a plugin. But I decided against it because of a couple problems I encountered:
1. Our notification system assumes that all push notifications refer to a post. So if we accept a raw `model.PostNotification` we would be putting it on the plugin to make sure the ids are all set correctly. This way, the plugin just has to know "valid Post", "valid Channel", and a target userID, and leaves the rest to the system (including following the server's rules for the message, given licensing and cfg settings).
2. The similar problem to https://github.com/mattermost/mattermost/pull/24263#discussion_r1297090335 -- the plugin probably shouldn't be allowed to send raw message data if the server has `IdLoadedNotification` set.
 
Basically: The way the PR does it, it allows a plugin to "suggest" a push notification, but the server will process it and make it conform to what is allowed, based on cfg settings and licensing.

Anyway, that's my thinking. Open to alternatives!

Reply:
Thanks, @cpoile! I like the framing of suggesting a push notification as it more clearly captures the intent. If we could find a name that captures this intent (leaving `sendPushNotification` for some future, more powerful API addition), I think my concerns would largely disappear.

Reply:
As discussed offline, moving to `SendPluginPushNotification` is a better indication for what it does. Thanks!